### PR TITLE
Open_Source_LLM Tool - Restore Dynamic List + Fixes

### PR DESCRIPTION
**Pull request by gjwoods**
# Description

Changes:
1. Restore Open_Source_LLM Tool Dynamic list functionality
2. Fixes: WS: 2806189, 2806307

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [x] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by chenslucky** [comment]
Please kindly revert your changes based on the latest main and try to avoid unnecessary changes to common code. The purpose is to try not to affect other tools.

File: src/promptflow-tools/tests/conftest.py [hunk 1]
```diff
@@ -79,18 +66,17 @@ def llama_chat_custom_connection():
...
         subs = json.loads(creds_custom_connection.secrets['service_credential'])
         for key, value in subs.items():
             os.environ[key] = value
         return True
     except Exception as e:
-        message = f"""=== Skipping Tests ===
-Something failed setting environment variables for service credentials. Error: {e}"""
-        print(message)
-        pytest.skip(message)
+        print(f"""Something failed setting environment variables for service credentials.
```
[the remark this answers is not in the document]
Comment (gjwoods) [reply src/promptflow-tools/tests/conftest.py:77]: This is the correct code.
The skip is in the test_open_source... file

**Comment by gjwoods** [comment]
> Please kindly revert your changes based on the latest main and try to avoid unnecessary changes to common code. The purpose is to try not to affect other tools.

**Comment by gjwoods** [comment]
> Please kindly revert your changes based on the latest main and try to avoid unnecessary changes to common code. The purpose is to try not to affect other tools.

Can you share more clarity on what you mean by this @chenslucky?
I've removed the changes from the common.py file.
For the revert of my changes - I don't follow? Are you asking me to rebase on the current main, or something else?

**Comment by chenslucky** [comment]
> Can you share more clarity on what you mean by this @chenslucky? I've removed the changes from the common.py file. For the revert of my changes - I don't follow? Are you asking me to rebase on the current main, or something else?

I just saw that some codes looked like they were not changed on the latest code. Some codes seem to have been reverted back to their previous modifications. But after your confirmation, now I have no doubts and the current code seems to have no problems. Thanks.

**Review by chenslucky — APPROVED** [review]

File: docs/reference/tools-reference/open_source_llm_tool.md [hunk 2]
```diff
@@ -2,48 +2,65 @@
 
 ## Introduction
```
Comment (chenslucky) [inline docs/reference/tools-reference/open_source_llm_tool.md:3]: Please add doc to docs/reference/index.md

Comment (gjwoods) [reply docs/reference/tools-reference/open_source_llm_tool.md:3]: I see it here: https://github.com/microsoft/promptflow/blob/main/docs/reference/index.md line 40.

**Review by DaweiCai — APPROVED** [review]

**Pull request merged**